ports/libopenmpt/CMakeLists.txt

```
@@ -44,224 +27,75 @@ include_directories(
-  soundlib/WAVTools.cpp
-  soundlib/WindowedFIR.cpp
-  soundlib/XMTools.cpp
-  soundlib/load_j2b.cpp
-  soundlib/mod_specifications.cpp
-  soundlib/modcommand.cpp
-  soundlib/modsmp_ctrl.cpp
-  soundlib/pattern.cpp
-  soundlib/patternContainer.cpp
-  soundlib/plugins/DigiBoosterEcho.cpp
-  soundlib/plugins/LFOPlugin.cpp
-  soundlib/plugins/PlugInterface.cpp
-  soundlib/plugins/PluginManager.cpp
-  soundlib/plugins/dmo/Chorus.cpp
-  soundlib/plugins/dmo/Compressor.cpp
-  soundlib/plugins/dmo/DMOPlugin.cpp
-  soundlib/plugins/dmo/Distortion.cpp
-  soundlib/plugins/dmo/Echo.cpp
-  soundlib/plugins/dmo/Flanger.cpp
-  soundlib/plugins/dmo/Gargle.cpp
-  soundlib/plugins/dmo/I3DL2Reverb.cpp
-  soundlib/plugins/dmo/ParamEq.cpp
-  soundlib/plugins/dmo/WavesReverb.cpp
-  soundlib/tuning.cpp
-  soundlib/tuningCollection.cpp
-  soundlib/tuningbase.cpp
-)
-
-add_library(libopenmpt ${lib_sources})
-
-target_compile_definitions(libopenmpt PRIVATE
-  -DMPT_WITH_MPG123 -DMPT_WITH_OGG
-  -DMPT_WITH_VORBIS -DMPT_WITH_VORBISFILE
-  -DMPT_WITH_ZLIB -DMPT_BUILD_VCPKG
-  -DLIBOPENMPT_BUILD
 )
 
-set(cli_srcs openmpt123/openmpt123.cpp)
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
-add_executable(openmpt123 ${cli_srcs})
+add_library(libopenmpt ${SOURCES})
 
-target_compile_definitions(openmpt123 PRIVATE
+target_compile_definitions(libopenmpt PRIVATE
   -DMPT_WITH_MPG123 -DMPT_WITH_OGG
   -DMPT_WITH_VORBIS -DMPT_WITH_VORBISFILE
   -DMPT_WITH_ZLIB -DMPT_BUILD_VCPKG
   -DLIBOPENMPT_BUILD
 )
 
-
-
 if(BUILD_SHARED_LIBS)
   target_compile_definitions(
     libopenmpt
     PRIVATE
-    -DMPT_BUILD_MSVC_SHARED -DLIBOPENMPT_BUILD_DLL
-  )
-  target_compile_definitions(
-    openmpt123
-    PRIVATE
-    -DMPT_BUILD_MSVC_SHARED -DLIBOPENMPT_BUILD_DLL
+      -DMPT_BUILD_MSVC_SHARED -DLIBOPENMPT_BUILD_DLL
   )
 else()
   target_compile_definitions(
     libopenmpt
     PRIVATE
-    -DMPT_BUILD_MSVC_STATIC
-  )
-  target_compile_definitions(
-    openmpt123
-    PRIVATE
-    -DMPT_BUILD_MSVC_STATIC
+      -DMPT_BUILD_MSVC_STATIC
   )
 endif()
 
-target_link_libraries(
+target_include_directories(
   libopenmpt
-  PRIVATE
-  MPG123::libmpg123
-  Vorbis::vorbisfile
-  FLAC::FLAC
-  ZLIB::ZLIB
+  PUBLIC
+    $<INSTALL_INTERFACE:include>
 )
 
-if(BUILD_SHARED_LIBS)
-  set(PORTAUDIO_TARGET portaudio)
-else()
-  set(PORTAUDIO_TARGET portaudio_static)
-endif()
-
 target_link_libraries(
-  openmpt123
-  PRIVATE
   libopenmpt
-  Winmm
-  ZLIB::ZLIB
-  ${PORTAUDIO_TARGET}
-)
-
-install(
-  TARGETS libopenmpt
-  RUNTIME DESTINATION bin
-  LIBRARY DESTINATION lib
-  ARCHIVE DESTINATION lib
+  PRIVATE
+    MPG123::libmpg123
+    Vorbis::vorbisfile
+    ZLIB::ZLIB
 )
 
-if(NOT DISABLE_INSTALL_HEADERS)
-  install(FILES ${lib_headers} DESTINATION include/libopenmpt)
-endif()
-
-if(NOT DISABLE_INSTALL_TOOLS)
-  install (
-    TARGETS openmpt123
-    RUNTIME DESTINATION tools/libopenmpt
-  )
-endif()
+set(LIBOPENMPT_REQUIRES_PRIVATE "zlib vorbisfile libmpg123")
```

Maybe this needs to be 
```suggestion
set(LIBOPENMPT_REQUIRES_PRIVATE "zlib vorbis vorbisfile libmpg123")
```

---

Yes, kind of. libopenmpt currently builds a function that calls `vorbis_comment_query()` (which is in libvorbis), however this function is unused in libopenmpt currently. In releases builds, function-level-linking and/or link-time-code-generation is smart enough to completely remove that code, however, in debug builds, linking libvorbis is currently required. I created upstream issue <https://bugs.openmpt.org/view.php?id=1477> to clean that up.